p2p/src/lib.rs

```diff
@@ -4,38 +4,144 @@
 //!
 //! This module defines support for (de)serialization and network transport
 //! of Bitcoin data and Bitcoin p2p network messages.
+extern crate alloc;
 
-#[cfg(feature = "std")]
 pub mod address;
-#[cfg(feature = "std")]
 pub mod message;
-#[cfg(feature = "std")]
 pub mod message_blockdata;
-#[cfg(feature = "std")]
 pub mod message_bloom;
-#[cfg(feature = "std")]
 pub mod message_compact_blocks;
-#[cfg(feature = "std")]
 pub mod message_filter;
-#[cfg(feature = "std")]
 pub mod message_network;
 
 use core::str::FromStr;
 use core::{fmt, ops};
 
 use hex::FromHex;
-use internals::{impl_to_hex_from_lower_hex, write_err};
+use internals::impl_to_hex_from_lower_hex;
 use io::{BufRead, Write};
 
-use crate::consensus::encode::{self, Decodable, Encodable};
-use crate::network::{Network, Params, TestnetVersion};
-use crate::prelude::{Borrow, BorrowMut, String, ToOwned};
+use bitcoin::consensus::encode::{self, Decodable, Encodable, WriteExt};
+use bitcoin::network::{Network, Params, TestnetVersion};
+use std::borrow::{Borrow, BorrowMut, ToOwned};
 
 #[rustfmt::skip]
 #[doc(inline)]
-#[cfg(feature = "std")]
 pub use self::address::Address;
 
+#[rustfmt::skip]
+#[allow(unused_imports)]
+mod prelude {
+    pub use std::{string::{String, ToString}, vec::Vec, boxed::Box, borrow::{Borrow, BorrowMut, Cow, ToOwned}, rc, sync};
+
+    pub use std::collections::{BTreeMap, BTreeSet, btree_map, BinaryHeap};
+
+    pub use io::sink;
+
+    pub use hex::DisplayHex;
+}
+
+pub(crate) fn consensus_encode_with_size<W: Write + ?Sized>(
```

Comment:
Maybe it would be cleaner to throw all this encoding stuff in a separate module?